types/react-native/index.d.ts

```diff
@@ -2033,16 +2033,52 @@ export interface AccessibilityProps extends AccessibilityPropsAndroid, Accessibi
     accessibilityRole?: AccessibilityRole;
     /**
      * Accessibility State tells a person using either VoiceOver on iOS or TalkBack on Android the state of the element currently focused on.
+     * deprecated: accessibilityState available in 0.60+
      */
-    accessibilityStates?: AccessibilityState[];
-
+    accessibilityStates?: AccessibilityStates[];
+    /**
+     * Accessibility State tells a person using either VoiceOver on iOS or TalkBack on Android the state of the element currently focused on.
+     */
+    accessibilityState?: AccessibilityState;
     /**
      * An accessibility hint helps users understand what will happen when they perform an action on the accessibility element when that result is not obvious from the accessibility label.
      */
     accessibilityHint?: string;
 }
 
-export type AccessibilityState = "selected" | "disabled";
+// deprecated: use AccessibilityState available in 0.60+
```

Comment:
I’m not sure how helpful it will be, but perhaps let’s try using the JSDoc annotation `@deprecated` so that it may trigger lint rules?

* https://jsdoc.app/tags-deprecated.html
* https://palantir.github.io/tslint/rules/deprecation/
* https://github.com/palantir/tslint/pull/2395

Reply:
I wasn't able to get tslint to report any warnings in our project consuming this, but the comment shows through with deprecated in a more visible color in VSCode

Reply:
(updated to @deprecated)

Reply:
Works for me, thanks!